lens-cube/src/main/java/org/apache/lens/cube/parse/StorageUtil.java

```diff
@@ -161,4 +165,96 @@ public static String getWhereClause(CandidateDim dim, String alias) {
       return null;
     }
   }
+
+  public static String getStorageTableName(Configuration conf, CubeFactTable fact, String storage) {
+    String str = conf.get(CubeQueryConfUtil.getValidStorageTablesKey(fact.getName()));
+    List<String> validFactStorageTables = StringUtils.isBlank(str)
+                                          ? null
+                                          : Arrays.asList(StringUtils.split(str.toLowerCase(), ","));
+    String tableName = getFactOrDimtableStorageTableName(fact.getName(), storage).toLowerCase();
+    if (validFactStorageTables != null && !validFactStorageTables.contains(tableName)) {
+      return null;
+    }
+    return tableName;
+  }
+
+  public static TimeRange getFallbackRange(TimeRange range, String factName, CubeQueryContext cubeql)
```

Comment:
👍  for moving all util like methods out of resolver and candidates to this util .